pkg/envoy/lds/inmesh.go

```diff
@@ -310,6 +330,59 @@ func buildTCPLocalRateLimitFilter(config *policyv1alpha1.TCPLocalRateLimitSpec,
 	return filter, nil
 }
 
+func buildTCPGlobalRateLimitFilter(config *policyv1alpha1.TCPGlobalRateLimitSpec, statPrefix string) (*xds_listener.Filter, error) {
+	if config == nil {
+		return nil, nil
+	}
+
+	rateLimit := &xds_global_ratelimit.RateLimit{
+		StatPrefix: statPrefix,
+		Domain:     config.Domain,
+		RateLimitService: &xds_config_ratelimit.RateLimitServiceConfig{
+			GrpcService: &xds_core.GrpcService{
+				TargetSpecifier: &xds_core.GrpcService_EnvoyGrpc_{
+					EnvoyGrpc: &xds_core.GrpcService_EnvoyGrpc{
+						ClusterName: service.RateLimitServiceClusterName(config.RateLimitService),
+					},
+				},
+			},
+			TransportApiVersion: xds_core.ApiVersion_V3,
+		},
+	}
+
+	var descriptors []*xds_common_ratelimit.RateLimitDescriptor
+	for _, desc := range config.Descriptors {
+		var entries []*xds_common_ratelimit.RateLimitDescriptor_Entry
+		for _, entry := range desc.Entries {
+			entries = append(entries, &xds_common_ratelimit.RateLimitDescriptor_Entry{Key: entry.Key, Value: entry.Value})
+		}
+
+		descriptors = append(descriptors, &xds_common_ratelimit.RateLimitDescriptor{Entries: entries})
+	}
+	rateLimit.Descriptors = descriptors
+
+	if config.Timeout != nil {
+		rateLimit.Timeout = durationpb.New(config.Timeout.Duration)
+		rateLimit.RateLimitService.GrpcService.Timeout = durationpb.New(config.Timeout.Duration)
+	}
+
+	if config.FailOpen != nil {
```

Comment:
nit: we could make this not a pointer, and use a default in the yaml definition for the CRD

Reply:
That's fine as well. I used this approach to be consistent with other optional fields for which we rely on Envoy defaults.